apps/src/aichat/views/AichatView.tsx

```diff
@@ -113,7 +117,10 @@ const AichatView: React.FunctionComponent = () => {
         hideForParticipants: true,
       })
     );
-  }, [dispatch, initialSources, levelAichatSettings]);
+    if (signInState === SignInState.SignedIn) {
+      dispatch(fetchUserHasAichatAccess());
+    }
```

Comment:
Also, might as well just call the API directly here rather than creating a whole separate async thunk? I don't think we would be calling this thunk function from any other component right? In which case we don't get much benefit from splitting it out into a separate function. For example this could be simplified to just:

```
useEffect(() => {
  if (signInState === SignInState.SignedIn) {
    fetchUserHasAichatAccess().then(hasAccess => dispatch(setUserHasAichatAccess(hasAccess));
  }
}, [signInState, dispatch]);
```
(also note that i'm only using promise syntax here because useEffect hooks can't be async functions. If you want to use async/await, you can create a separate async function and call that within the useEffect hook, but that may be overkill).

Reply:
Makes sense - thanks! Updated.